realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -469,13 +466,11 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 query.notifyChangeListeners();
             }
 
-            // notify listeners only when we advanced
-            if (compare != 0) {
-                notifyGlobalListeners();
-                // notify RealmResults & RealmObject callbacks (type based notifications)
-                notifySyncRealmResultsCallbacks();
-                notifyRealmObjectCallbacks();
-            }
+            // We need to notify the rest of listeners, since the original REALM_CHANGE
+            // was delayed/swallowed in order to be able to update async queries
```

Comment:
:+1:  Just figured out this could happen when the transaction is committed in the same thread which has async queries.